src/test/java/org/terasology/web/serverAdminManagement/AdminPermissionManagerTest.java

```diff
@@ -0,0 +1,134 @@
+
+public class AdminPermissionManagerTest {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private AdminPermissionManager createPermissionListWithTwoAdmins() {
+        AdminPermissionManager adminPermissionManager = new AdminPermissionManager(null, false);
+        adminPermissionManager.addAdmin("first");
+        adminPermissionManager.giveAllPermissionsToAdmin("first");
+        adminPermissionManager.addAdmin("second");
+        return adminPermissionManager;
+    }
+
+    @Test
+    public void testAdminAddRemove() {
+        AdminPermissionManager adminPermissionManager = new AdminPermissionManager(null, false);
+        assertTrue(adminPermissionManager.getAdminPermissions().isEmpty());
+        adminPermissionManager.addAdmin("test");
+        assertEquals(adminPermissionManager.getAdminPermissions().iterator().next().getId(), "test");
+        assertFalse(adminPermissionManager.getAdminPermissions().isEmpty());
+        adminPermissionManager.removeAdmin("test");
+        assertTrue(adminPermissionManager.getAdminPermissions().isEmpty());
+    }
+
+    @Test
+    public void testFirstAdminHasAllPermissions() {
+        AdminPermissionManager adminPermissionManager = createPermissionListWithTwoAdmins();
+        Map<PermissionType, Boolean> allPermissionsTrue = PermissionType.generatePermissionMap(true);
+        assertEquals(allPermissionsTrue, adminPermissionManager.getPermissionsOfAdmin("first"));
+    }
+
+    @Test
+    public void testSecondAdminHasNoPermissions() {
+        AdminPermissionManager adminPermissionManager = createPermissionListWithTwoAdmins();
+        Map<PermissionType, Boolean> allPermissionsFalse = PermissionType.generatePermissionMap(false);
+        assertEquals(allPermissionsFalse, adminPermissionManager.getPermissionsOfAdmin("second"));
+    }
+
+    // TODO make work
+    @Test
+    @Ignore
+    public void testConsoleAdminPermissions() {
+        PermissionManager permissionManagerMock = mock(PermissionManager.class);
+        EntityManager entityManagerMock = mock(EntityManager.class);
+        EntityRef clientEntityRefMock = mock(EntityRef.class);
+        when(entityManagerMock.create("engine:client")).thenReturn(clientEntityRefMock);
+        when(clientEntityRefMock.getComponent(ClientComponent.class)).thenReturn(new ClientComponent());
+
+        Context context = new ContextImpl();
+        context.put(PermissionManager.class, permissionManagerMock);
+        AdminPermissionManager adminPermissionManager = createPermissionListWithTwoAdmins();
+        InjectionHelper.inject(adminPermissionManager, context);
+
+        // The permission manager mock does not correctly add the permissions
+        adminPermissionManager.updateAdminConsolePermissions("first", clientEntityRefMock);
+
+        // Therefore, this is false
+        assertTrue(permissionManagerMock.hasPermission(clientEntityRefMock, PermissionManager.CHEAT_PERMISSION));
```

Comment:
Any hints on why this doesn't work? It seems like either the mock isn't working correctly or I'm using the permission system incorrectly.